pennylane/allocation.py

```diff
@@ -209,47 +209,113 @@ def allocate(num_wires: int, require_zeros: bool = True, restored: bool = False)
     """Dynamically allocates new wires in-line or as a context manager, which also safely deallocates the
     new wires upon exiting the context.
 
-    .. warning::
-
-        This feature is experimental and is not possible on any device yet.
+    .. warning:: 
+        This feature is experimental, and any workflows that include calls to ``allocate`` cannot be 
+        executed on any device.
 
     Args:
-        num_wires (int): the number of dynamic wires to allocate.
+        num_wires (int): the number of wires to dynamically allocate.
 
     Keyword Args:
-        require_zeros (bool): whether or not the wires must start in the ``0`` state
-        restored (bool): whether or not the wires are returned to the same state they started in.
+        require_zeros (bool): 
+            specifies whether to allocate ``num_wires`` in the all-zeros state (``True``) or in any 
+            arbitrary state (``False``). The default value is ``require_zeros=True``.
+
+        restored (bool): 
+            whether or not the dynamically allocated wires are returned to the same state they started 
+            in. ``restored=True`` indicates that the user promises to restore the dynamically allocated 
+            memory to its original state before :func:`~.deallocate` is called. ``restored=False`` indicates 
+            that the user does not promise to restore the dynamically allocated memory before :func:`~.deallocate` 
+            is called. The default value is ``False``.
+
+    Returns:
+        DynamicRegister: an object, behaving similarly to ``Wires``, that represents the dynamically 
+        allocated memory.
+
+    .. note::
+        The ``allocate`` function should be used with :func:`~.deallocate` if it is not used as a context 
+        manager.
+
+    .. see-also::
+        :func:`~.deallocate`
+
+    **Examples**:
 
     This function can be used as a context manager with automatic deallocation (preferred) or with manual
     deallocation via :func:`~.deallocate`.
 
     .. code-block:: python
 
-        @qml.qnode(qml.device('default.qubit', wires=("a", "b")))
+        @qml.qnode(qml.device('default.qubit', wires=["a", "b"]))
```

Comment:
What's wrong with tuples?

Reply:
I honestly don't know hahaha I have an irrational aversion to tuples 🥴 

```suggestion
        @qml.qnode(qml.device('default.qubit', wires=("a", "b")))
```